tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/FunExceptRule.scala

```diff
@@ -103,7 +103,7 @@ class FunExceptRule(rewriter: SymbStateRewriter) extends RewritingRule {
     }
 
     // compute all updated cells in case we are dealing with a function over non-basic indices
-    val updatedCells = relationCells.map(eachRelationPair)
+    relationCells.map(eachRelationPair)
```

Comment:
check whether there are side effects we need here or if this can be removed.

Reply:
Yes, `eachRelationPair` has side effects -- it is submitting queries to the solver. It would be better to have `forEach` here.